### DISCUSSION TITLE
As of v0.1.0-alpha the code is functional.

### DISCUSSION THREAD

**Discussion by jorgejch** [jorgejch/dex-arbitrage#3] [Announcements]
As of version v0.1.0-alpha, the code is functional. However, in my testing environment, it is not generating any profit. The transactions are being reverted due to a lack of profitability. In summary, this indicates that it is losing in the competitive landscape.

I would appreciate any help optimizing for speed and checking for calculation errors.

Latest test contract: 0xC6B427650BA488c5702dA217713Ac782E58B60Bc

**Comment by milescarlston** [comment]
Is this issue only present with version v0.1.0-alpha or do you have a previous version that is profitably?

**Reply by jorgejch** [reply]
Nope, that's my first version. Profit is the holy grail. I've added some more optimization and will continue to do so, but I'll need to run my node to make fast calls. There are some issues with the gas, too, and that needs optimization as well.
